### PR TITLE
CON-2326: Search route has display bugs

### PR DESCRIPTION
**Pull request by richardgorman**

### CODE REVIEW

File: filter/subscriptions/index.js [hunk 1]
```diff
@@ -21,33 +22,43 @@ import { SEARCH_PATH } from '../../search/constants';
  */
 export default function filters(subscribe) {
   /**
+   * Gets triggered when entering the filter route.
```
Comment (coldi) [inline filter/subscriptions/index.js]: "... when leaving ..."

File: filter/subscriptions/index.js [hunk 2]
```diff
@@ -21,33 +22,43 @@ import { SEARCH_PATH } from '../../search/constants';
...
+   */
+  const filterRouteDidLeave$ = routeDidLeave(FILTER_PATH);
+
+  /**
+   * Gets triggered when entering the filter route.
+   */
+  const filterRouteDidEnter$ = routeDidEnter(FILTER_PATH);
```
Comment (coldi) [inline filter/subscriptions/index.js]: I lately experienced that creation order of streams does matter.
Therefore and for code structural reasons you may want to create `...Enter$` before `...Leave$`.

File: search/streams/index.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+export const searchRequesting$ = main$.filter(
+  ({ action }) => action.type === REQUEST_SEARCH_RESULTS
+);
+
+/**
+ * Gets triggered when search results are received.
+ * @type {Observable}
+ */
+export const searchReceived$ = main$.filter(
+  ({ action }) =>
```
Comment (coldi) [inline search/streams/index.js]: Please remove line break and use same format as above.

File: search/subscriptions/index.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+import setViewLoading from '@shopgate/pwa-common/actions/view/setViewLoading';
+import unsetViewLoading from '@shopgate/pwa-common/actions/view/unsetViewLoading';
+import {
+  searchRequesting$,
+  searchReceived$,
+} from '../streams';
+import { SEARCH_PATH } from '../constants';
+
+/**
+ * Cart subscriptions.
```
Comment (coldi) [inline search/subscriptions/index.js]: "Search ..."

File: search/subscriptions/index.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+ * @param {Function} subscribe The subscribe function.
+ */
+export default function search(subscribe) {
+  subscribe(searchRequesting$, ({ dispatch }) => {
+    dispatch(setViewLoading(SEARCH_PATH));
```
Comment (coldi) [inline search/subscriptions/index.js]: You may consider moving this file to a more general "UI" feature.
Then the UI would subscribe to streams of the search feature.
Therefore search does not need a dependency to UI.
Makes more sense to me.

File: filter/subscriptions/index.js (hunk 1 above)
Comment (richardgorman) [reply filter/subscriptions/index.js]: Corrected.

File: search/streams/index.js (hunk 3 above)
Comment (richardgorman) [reply search/streams/index.js]: Corrected.

File: search/subscriptions/index.js (hunk 4 above)
Comment (richardgorman) [reply search/subscriptions/index.js]: Corrected.

File: search/subscriptions/index.js (hunk 5 above)
Comment (richardgorman) [reply search/subscriptions/index.js]: I won't do this because this is not strictly related to the UI. These actions set flags that the UI can react upon.

Comment (coldi) [reply search/subscriptions/index.js]: Okay

File: filter/subscriptions/index.js (hunk 2 above)
Comment (richardgorman) [reply filter/subscriptions/index.js]: Done, Enter and Leave swapped order.

**Pull request merged**